### DISCUSSION TITLE
Rust rls & rust-toolstate: nightly-2021-01-07

### DISCUSSION THREAD

**Discussion by wright-joe** [tock/tock#2485] [Q&A] [answered]
Hello. 

I am spinning up a project, and noticed the toolchain is set to 'nightly-2021-01-07'.  This causes a conflict with the 'rls' component installation, as it is not part of that specific build.

> error: component 'rls' for target 'x86_64-unknown-linux-gnu' is unavailable for download for channel nightly-2021-01-07
> Sometimes not all components are available in any given nightly.

Does anyone have a workaround they recommend?

**Comment by bradjc** [comment, marked as answer]
You should have no problem just choosing a newer nightly that has the rls component. Tock pegs to a specific nightly for convenience, but we typically don't see any issues with newer nightlies.

One thing I don't understand is that our script that helps us select nightlies with the required components:

https://github.com/tock/tock/blob/master/tools/update_rust_version.sh#L7

does specify we want rls support. So I'm not sure why we ended up on a nightly that doesn't include it.

**Reply by wright-joe** [reply]
Thank you.

`"rust-client.channel": "nightly-2021-03-09",` 

seems to be working well so far.
